lib/Translation/EmitHLSCpp.cpp

```diff
@@ -1880,13 +1949,27 @@ void ModuleEmitter::emitModule(ModuleOp module) {
 #include <math.h>
 #include <stdint.h>
 #include <string.h>
-
-using namespace std;
-
 )XXX";
 
-  // Emit all functions in the call graph in a post order.
+  // Emit all includes for library ip.
+  module->walk([&](Operation *op) {
+    if (auto libraryOp = dyn_cast<hls::LibraryOp>(op)) {
+      libraryOp.walk([&](Operation *nestedOp) {
+        if (nestedOp->getName().getStringRef() == "hls.uip.include") {
+          auto curPath = nestedOp->getAttr("paths").dyn_cast<ArrayAttr>()[0];
+          os << "#include ";
+          os << curPath;
+          os << "\n";
+        }
+      });
+    }
+  });
```

Comment:
Instead of printing the header of all IPs, we should only print the IPs that are used in the current design?